tools/ci/ci_agent.py

```diff
@@ -278,6 +312,68 @@ def _run_targets(targets, src, build_path, config):
     return result
 
 
+def _build_code_coverage(src, build_path, build_flags, c_flags, codecov_token):
```

Comment:
what's the significance of `build_flags` vs `c_flags`?

Reply:
in csdk we have 2 levels of flags one for cmake and other for compiler

build_flags is for cmake and c_flags are for compiler

Reply:
I don't like the build_flags name. But at the very least we should have a descriptive doc string saying what they are.